### PR TITLE
feat(users): add endpoint for terminate auth select

### LINKED ISSUE #5127: feat: add endpoint for terminate auth select
Add support to terminate auth select, 
- the api will take Auth Select SPT and
- it will either SSO or TOTP SPT as per requested scenario

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Add endpoint `/auth/select` to terminate auth select
- The api will take Auth Select SPT and will give SSO SPT or TOTP SPT based on the choice selected
- SSO SPT if user selects continue with SSO
- TOTP SPT if user selects continue with TOTP

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Closes #5127 

## How did you test it?
If request from id that has auth type as `open_id_connect`:
```
curl --location 'http://localhost:8080/user/auth/select' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer SPT' \
--header 'Cookie: Cookie_1=value' \
--data '{
    "id":"f20742d2-f1b8-4dad-8b2d-cf8fe6d457eb"

}'
```
response will contain token type as `sso`
```
{
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYjJkNjMxZDUtMGQzMi00Y2IxLTg2MTQtNWVlNWI2MDczYTVkIiwicHVycG9zZSI6InNzbyIsIm9yaWdpbiI6ImFjY2VwdF9pbnZpdGF0aW9uX2Zyb21fZW1haWwiLCJwYXRoIjpbImF1dGhfc2VsZWN0Il0sImV4cCI6MTcxOTU4NDc2NX0.ChD9An8L4szVjLn9Q51otVOXOuZf8kdgRMmH865lQQI",
    "token_type": "sso"
}
```
If request from id other than `open_id_connect` 
```
curl --location 'http://localhost:8080/user/auth/select' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer SPT' \
--header 'Cookie: Cookie_1=value' \
--data '{

    "id":"65ff02eb-f3dc-4afc-b8ac-cef2a30c4d12"

}'
```
Response will contain token type as `totp`
```
{
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYjJkNjMxZDUtMGQzMi00Y2IxLTg2MTQtNWVlNWI2MDczYTVkIiwicHVycG9zZSI6InRvdHAiLCJvcmlnaW4iOiJhY2NlcHRfaW52aXRhdGlvbl9mcm9tX2VtYWlsIiwicGF0aCI6WyJhdXRoX3NlbGVjdCJdLCJleHAiOjE3MTk1ODQ5MDl9.hEBp49fycxiSW1VoyPR6s9TISO4lQKP3ta0S4SH3C9s",
    "token_type": "totp"
}
```
Db entries for correponding request ids:
Screenshot 2024-06-26 at 7 59 41 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/user.rs [hunk 1]
```diff
@@ -2312,3 +2312,41 @@ pub async fn sso_sign(
...
+
+    let current_flow = domain::CurrentFlow::new(user_token, domain::SPTFlow::AuthSelect.into())?;
+    let mut next_flow = current_flow.next(user_from_db.clone(), &state).await?;
+
+    // Skip SSO if continue with password(TOTP)
+    if next_flow.get_flow() == domain::UserFlow::SPTFlow(domain::SPTFlow::SSO)
+        && user_authentication_method.auth_type != common_enums::UserAuthType::OpenIdConnect
+    {
+        next_flow = next_flow.skip(user_from_db, &state).await?;
+    }
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:2342]: Can we use match instead.
```suggestion
next_flow = match (next_flow.get_flow(), user_authentication_method.auth_type) {
    (SSO, OpenIdConnect) => skip(),
    (_, Password) | (_, MagicLink => next_flow,
}
```

**Review by ThisIsMani — APPROVED** [review]

File: crates/router/src/core/user.rs (hunk 1 above)
Comment (apoorvdixit88) [reply crates/router/src/core/user.rs:2342]: we can cover this with other refactoring for sso (parsing etc)

File: crates/router/src/core/user.rs [hunk 2]
```diff
@@ -2312,3 +2312,41 @@ pub async fn sso_sign(
...
+        .find_user_by_id(&user_token.user_id)
+        .await
+        .change_context(UserErrors::InternalServerError)?
+        .into();
+
+    let user_authentication_method = state
+        .store
+        .get_user_authentication_method_by_id(&req.id)
+        .await
+        .change_context(UserErrors::InternalServerError)?;
```
Comment (racnan) [inline crates/router/src/core/user.rs:2332]: why will it be `InternalServerError` if id is not found ?

**Review by racnan — APPROVED** [review]

File: crates/router/src/core/user.rs (hunk 2 above)
Comment (apoorvdixit88) [reply crates/router/src/core/user.rs:2332]: we can change it, anyways id will be send from the response of list api, so it will be always valid.

**Pull request merged**